pkg/reconciler/subscription/subscription_test.go

```diff
@@ -2301,6 +2337,156 @@ func TestAllCases(t *testing.T) {
 			WantPatches: []clientgotesting.PatchActionImpl{
 				patchRemoveFinalizers(testNS, subscriptionName),
 			},
+		}, {
+			Name: "OIDC: creates OIDC service account",
+			Key:  testNS + "/" + subscriptionName,
+			Ctx: feature.ToContext(context.Background(), feature.Flags{
+				feature.OIDCAuthentication: feature.Enabled,
+			}),
+			Objects: []runtime.Object{
+				NewSubscription(subscriptionName, testNS,
+					WithSubscriptionUID(subscriptionUID),
+					WithSubscriptionChannel(imcV1GVK, channelName),
+					WithSubscriptionSubscriberRef(subscriberGVK, subscriberName, testNS),
+					WithSubscriptionReply(imcV1GVK, replyName, testNS),
+					WithInitSubscriptionConditions,
+					WithSubscriptionFinalizers(finalizerName),
+					MarkReferencesResolved,
+					MarkAddedToChannel,
+					WithSubscriptionPhysicalSubscriptionSubscriber(&subscriber),
+					WithSubscriptionPhysicalSubscriptionReply(&reply),
+				),
+				// Subscriber
+				NewUnstructured(subscriberGVK, subscriberName, testNS,
+					WithUnstructuredAddressable(subscriber),
+				),
+				// Reply
+				NewInMemoryChannel(replyName, testNS,
+					WithInitInMemoryChannelConditions,
+					WithInMemoryChannelAddress(reply),
+				),
+				// Channel
+				NewInMemoryChannel(channelName, testNS,
+					WithInitInMemoryChannelConditions,
+					WithInMemoryChannelReady(channelDNS),
+					WithInMemoryChannelSubscribers([]eventingduck.SubscriberSpec{{
+						UID:           subscriptionUID,
+						Generation:    0,
+						SubscriberURI: subscriberURI,
+						ReplyURI:      replyURI,
+					}, {
+						UID:           "34c5aec8-deb6-11e8-9f32-f2801f1b9fd1",
+						Generation:    1,
+						SubscriberURI: apis.HTTP("call2"),
+						ReplyURI:      apis.HTTP("sink2"),
```

Comment:
This seems unnecessary for this test too. Maybe something like the following as a starting point?

```
Name: "OIDC: creates OIDC service account",
Key:  testNS + "/" + subscriptionName,
Ctx: feature.ToContext(context.Background(), feature.Flags{
	feature.OIDCAuthentication: feature.Enabled,
}),
Objects: []runtime.Object{
	NewSubscription(subscriptionName, testNS,
		WithSubscriptionUID(subscriptionUID),
		WithSubscriptionChannel(imcV1GVK, channelName),
		WithSubscriptionSubscriberRef(subscriberGVK, subscriberName, testNS),
		WithInitSubscriptionConditions,
		WithSubscriptionFinalizers(finalizerName),
		MarkSubscriptionReady,
	),
	// Subscriber
	NewUnstructured(subscriberGVK, subscriberName, testNS,
		WithUnstructuredAddressable(subscriber),
	),
	// Channel
	NewInMemoryChannel(channelName, testNS,
		WithInitInMemoryChannelConditions,
		WithInMemoryChannelReady(channelDNS),
		WithInMemoryChannelSubscribers([]eventingduck.SubscriberSpec{{
			SubscriberURI: subscriberURI,
			UID:           subscriptionUID,
		}}),
		WithInMemoryChannelAddress(channelDNS),
		WithInMemoryChannelStatusSubscribers([]eventingduck.SubscriberStatus{{
			UID: subscriptionUID,
		}}),
	),
},
WantErr: false,
WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
	Object: NewSubscription(subscriptionName, testNS,
		WithSubscriptionUID(subscriptionUID),
		WithSubscriptionChannel(imcV1GVK, channelName),
		WithSubscriptionSubscriberRef(subscriberGVK, subscriberName, testNS),
		WithInitSubscriptionConditions,
		WithSubscriptionFinalizers(finalizerName),
		WithSubscriptionPhysicalSubscriptionSubscriber(&subscriber),
		// - Status Update -
		MarkSubscriptionReady,
		WithSubscriptionOptionOIDCIdentityCreatedSucceeded(),
		WithSubscriptionOptionOIDCServiceAccountName(makeSubscriptionOIDCServiceAccount().Name),
	),
}},
WantCreates: []runtime.Object{
	makeSubscriptionOIDCServiceAccount(),
},
```

I usually try to copy the simplest test and adjust this for my needs :shrug: